scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/Conversion.scala

```diff
@@ -0,0 +1,215 @@
+package pl.touk.nussknacker.engine.extension
+
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypingResult, Unknown}
+import pl.touk.nussknacker.engine.extension.Conversion.toNumber
+import pl.touk.nussknacker.engine.spel.internal.ConversionHandler
+import pl.touk.nussknacker.engine.util.classes.Extensions.ClassExtensions
+
+import scala.util.{Success, Try}
+import java.util.{
+  ArrayList => JArrayList,
+  Collection => JCollection,
+  HashMap => JHashMap,
+  List => JList,
+  Map => JMap,
+  Set => JSet
+}
+import java.lang.{Boolean => JBoolean, Double => JDouble, Long => JLong, Number => JNumber}
+import java.math.{BigDecimal => JBigDecimal, BigInteger => JBigInteger}
+
+sealed trait Conversion {
+  type ResultType >: Null <: AnyRef
+  val resultTypeClass: Class[ResultType]
+
+  def convertEither(value: Any): Either[Throwable, ResultType]
+  def applies(clazz: Class[_]): Boolean
+
+  def canConvert(value: Any): JBoolean = convertEither(value).isRight
+
+  def convert(value: Any): ResultType = convertEither(value) match {
+    case Right(value) => value
+    case Left(ex)     => throw ex
+  }
+
+  def convertOrNull(value: Any): ResultType = convertEither(value) match {
+    case Right(value) => value
+    case Left(_)      => null
+  }
+
+  def typingResult: TypingResult = Typed.typedClass(resultTypeClass)
+}
+
+sealed trait NumericConversion extends Conversion {
+  private val numberClass  = classOf[JNumber]
+  private val stringClass  = classOf[String]
+  private val unknownClass = classOf[Object]
+
+  override def applies(clazz: Class[_]): Boolean =
+    clazz.isAOrChildOf(numberClass) || clazz == stringClass || clazz == unknownClass
```

Comment:
Don't we need `clazz != resultTypeClass && (...)` to not present redundant `1.toInteger`?

Reply:
Done